playlist.c

```diff
@@ -1502,21 +1528,42 @@ void playlist_write_file(
          json_write_space(context.writer, 4);
          JSON_Writer_WriteStartObject(context.writer);
 
+         char* path = string_is_empty(playlist->entries[i].relative_path) ? playlist->entries[i].path : NULL;
+
          json_write_new_line(context.writer);
          json_write_space(context.writer, 6);
          JSON_Writer_WriteString(context.writer, "path",
                STRLEN_CONST("path"), JSON_UTF8);
          JSON_Writer_WriteColon(context.writer);
          json_write_space(context.writer, 1);
          JSON_Writer_WriteString(context.writer,
-               playlist->entries[i].path
-               ? playlist->entries[i].path
```

Comment:
No, leave the path stuff here alone. For C89 compatibility you have to declare variables at the start of the scope, so your `char* path` will end up miles away, and the code will be unclear...